Warning found by clang --analyze

This might not be an issue, however, I saw the warning and thought I report it here anyway, just in case it is useful to you. 

```
+clang --analyze -w -DNNG_ENABLE_STATS -DNNG_HAVE_ALLOCA=1 -DNNG_HAVE_BACKTRACE=1 -DNNG_HAVE_BUS0 -DNNG_HAVE_CLOCK_GETTIME=1 -DNNG_HAVE_EPOLL=1 -DNNG_HAVE_EPOLL_CREATE1=1 -DNNG_HAVE_FLOCK=1 -DNNG_HAVE_LIBNSL=1 -DNNG_HAVE_LOCKF=1 -DNNG_HAVE_MSG_CONTROL=1 -DNNG_HAVE_PAIR0 -DNNG_HAVE_PAIR1 -DNNG_HAVE_PUB0 -DNNG_HAVE_PULL0 -DNNG_HAVE_PUSH0 -DNNG_HAVE_REP0 -DNNG_HAVE_REQ0 -DNNG_HAVE_RESPONDENT0 -DNNG_HAVE_SEMAPHORE_PTHREAD=1 -DNNG_HAVE_SOPEERCRED=1 -DNNG_HAVE_STDATOMIC=1 -DNNG_HAVE_STRCASECMP=1 -DNNG_HAVE_STRNCASECMP=1 -DNNG_HAVE_STRNLEN=1 -DNNG_HAVE_SUB0 -DNNG_HAVE_SURVEYOR0 -DNNG_HAVE_UNIX_SOCKETS=1 -DNNG_HIDDEN_VISIBILITY -DNNG_LITTLE_ENDIAN -DNNG_PLATFORM_LINUX -DNNG_PLATFORM_POSIX -DNNG_PRIVATE -DNNG_STATIC_LIB -DNNG_SUPP_HTTP -DNNG_TRANSPORT_INPROC -DNNG_TRANSPORT_IPC -DNNG_TRANSPORT_TCP -DNNG_TRANSPORT_WS -DNNG_USE_EVENTFD -D_GNU_SOURCE -D_POSIX_PTHREAD_SEMANTICS -D_REENTRANT -D_THREAD_SAFE -isystem /home/travis/build/tnixeu/hunter_old/_testing/Hunter/_Base/c39ed97/164fd52/f0e3d86/Build/nng/Source/src -Wall -Wextra -fno-omit-frame-pointer -O3 -DNDEBUG -std=gnu99 -o CMakeFiles/nng.dir/core/message.c.o -c /home/travis/build/tnixeu/hunter_old/_testing/Hunter/_Base/c39ed97/164fd52/f0e3d86/Build/nng/Source/src/core/message.c -o /tmp/polly-clang-analyze.bin.TYtmf
+RESULT=0
+'[' 0 == 0 ']'
+'[' -s /tmp/polly-clang-analyze.out.t4waK ']'
+RESULT=1
+cat /tmp/polly-clang-analyze.out.t4waK
/home/travis/build/tnixeu/hunter_old/_testing/Hunter/_Base/c39ed97/164fd52/f0e3d86/Build/nng/Source/src/core/message.c:246:3: warning: Null pointer passed as an argument to a 'nonnull' parameter
                memcpy(ch->ch_ptr + ch->ch_len, data, len);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

It is a warning found by `clang --analyze`. I tried understanding the warning, however, I am not really used to reading C code. 

`clang --analyze` was run during the process of adding `nng` in version v1.1.1 to a package manager. When adding a package there, it is build by travis-ci. You can find the build log [here](https://travis-ci.org/tnixeu/hunter_old/jobs/602497181#L2385).

--- a/src/core/message.c
+++ b/src/core/message.c
@@ -140,7 +140,9 @@ nni_chunk_grow(nni_chunk *ch, size_t newsz, size_t headwanted)
 			return (NNG_ENOMEM);
 		}
 		// Copy all the data, but not header or trailer.
-		memcpy(newbuf + headwanted, ch->ch_ptr, ch->ch_len);
+		if (ch->ch_len > 0) {
+			memcpy(newbuf + headwanted, ch->ch_ptr, ch->ch_len);
+		}
 		nni_free(ch->ch_buf, ch->ch_cap);
 		ch->ch_buf = newbuf;
 		ch->ch_ptr = newbuf + headwanted;
@@ -221,7 +223,9 @@ nni_chunk_dup(nni_chunk *dst, const nni_chunk *src)
 	dst->ch_cap = src->ch_cap;
 	dst->ch_len = src->ch_len;
 	dst->ch_ptr = dst->ch_buf + (src->ch_ptr - src->ch_buf);
-	memcpy(dst->ch_ptr, src->ch_ptr, dst->ch_len);
+	if (dst->ch_len > 0) {
+		memcpy(dst->ch_ptr, src->ch_ptr, dst->ch_len);
+	}
 	return (0);
 }
 
@@ -278,7 +282,7 @@ nni_chunk_insert(nni_chunk *ch, const void *data, size_t len)
 	}
 
 	ch->ch_len += len;
-	if (data) {
+	if (data != NULL) {
 		memcpy(ch->ch_ptr, data, len);
 	}
 
@@ -466,7 +470,9 @@ nni_msg_dup(nni_msg **dup, const nni_msg *src)
 		newmo->mo_val = ((char *) newmo + sizeof(*newmo));
 		newmo->mo_sz  = mo->mo_sz;
 		newmo->mo_num = mo->mo_num;
-		memcpy(newmo->mo_val, mo->mo_val, mo->mo_sz);
+		if (mo->mo_sz > 0) {
+			memcpy(newmo->mo_val, mo->mo_val, mo->mo_sz);
+		}
 		nni_list_append(&m->m_options, newmo);
 	}
 	m->m_pipe = src->m_pipe;
@@ -500,7 +506,7 @@ nni_msg_setopt(nni_msg *m, int opt, const void *val, size_t sz)
 
 	NNI_LIST_FOREACH (&m->m_options, oldmo) {
 		if (oldmo->mo_num == opt) {
-			if (sz == oldmo->mo_sz) {
+			if (sz == oldmo->mo_sz && sz > 0) {
 				// nice! we can just overwrite old value
 				memcpy(oldmo->mo_val, val, sz);
 				return (0);
@@ -514,7 +520,9 @@ nni_msg_setopt(nni_msg *m, int opt, const void *val, size_t sz)
 	newmo->mo_val = ((char *) newmo + sizeof(*newmo));
 	newmo->mo_sz  = sz;
 	newmo->mo_num = opt;
-	memcpy(newmo->mo_val, val, sz);
+	if (sz > 0) {
+		memcpy(newmo->mo_val, val, sz);
+	}
 	if (oldmo != NULL) {
 		nni_list_remove(&m->m_options, oldmo);
 		nni_free(oldmo, sizeof(*oldmo) + oldmo->mo_sz);
@@ -533,7 +541,9 @@ nni_msg_getopt(nni_msg *m, int opt, void *val, size_t *szp)
 			size_t sz = *szp;
 			if (sz > mo->mo_sz) {
 				sz = mo->mo_sz;
-				memcpy(val, mo->mo_val, sz);
+				if (sz > 0) {
+					memcpy(val, mo->mo_val, sz);
+				}
 				*szp = mo->mo_sz;
 				return (0);
 			}

--- a/src/supplemental/websocket/websocket.c
+++ b/src/supplemental/websocket/websocket.c
@@ -846,12 +846,14 @@ ws_read_finish_str(nni_ws *ws)
 				// This eats the entire iov.
 				n = iov->iov_len;
 			}
-			memcpy(iov->iov_buf, frame->buf, n);
-			iov->iov_buf = ((uint8_t *) iov->iov_buf) + n;
-			iov->iov_len -= n;
-			if (iov->iov_len == 0) {
-				iov++;
-				niov--;
+			if (n != 0) {
+				memcpy(iov->iov_buf, frame->buf, n);
+				iov->iov_buf = ((uint8_t *) iov->iov_buf) + n;
+				iov->iov_len -= n;
+				if (iov->iov_len == 0) {
+					iov++;
+					niov--;
+				}
 			}
 
 			if (frame->len == n) {

